client/utils.go

```diff
@@ -90,7 +90,7 @@ func encodePlatforms(platform ...ocispec.Platform) ([]string, error) {
 func encodePlatform(platform *ocispec.Platform) (string, error) {
 	p, err := json.Marshal(platform)
 	if err != nil {
-		return "", errdefs.InvalidParameter(fmt.Errorf("invalid platform: %v", err))
+		return "", fmt.Errorf("%w: invalid platform: %v", cerrdefs.ErrInvalidArgument, err)
```

Comment:
Fine for now, but wondering if containerd/errdefs has a good way to type the error without that meaning that the literal `invalid argument` is part of the message 🤔

Reply:
I guess something like; 

```suggestion
		return "", cerrdefs.ErrInvalidArgument.WithMessage(fmt.Sprint("invalid platform: %v", err))
```

But that would discard the underlying error (which would be fine for this case, but perhaps not for others 🤔

Reply:
`WithMessage` is useful if you "need" a specific message, I would say "invalid argument: invalid platform" would be fine as it goes to more specific still. To keep the original error, multiple `%w` can be provided as well.

Reply:
Yeah, it's just that in many cases, adding the literal `some-error-type` to the message is just noise; at least for the user (these tend to be printed / presented to the user). Where it's relevant to _us_ to determine the error type for code flow, ideally that implementation detail shouldn't have to make its way to the user.


e.g. want to avoid things like;

    failed to create container: invalid-argument: network some-network was not found: not-found